### ISSUE TITLE
Bug: (fill in)Using underscore in a BigInt literal results in an error

### ISSUE THREAD

**Issue by JeffML** [eslint/eslint#15521]
### Environment

Node version: 14.17.5
npm version: 8.3.0
Local ESLint version: 6.8.0
Global ESLint version: none
Operating System: ChromeOS

### What parser are you using?

@babel/eslint-parser

### What did you do?

<details>
See code snippet below to reproduce
</details>

```js
let a = 100000n;  // this works; BigInt literal
let b = 100_000   //this works; integer literal
let c = 100_000n  // this fails; BigInt literal with underscore separator
```
Note that Chrome browser has not problem interpreting 100_000n as a big int.

### What did you expect to happen?

I expect ESLint not to complain about valid JS literals.

### What actually happened?

Got the following error:

> [eslint/eslint] Module parse failed: Identifier directly after number

### Participation

- [ ] I am willing to submit a pull request for this issue.

**Comment by aladdin-add** [comment]
It's working in the latest eslint and default parser.

can you upgrade your eslint and `@babel/eslint-parser` to the latest version, and retry? it would be an issue in `@babel/eslint-parser` if it still happens.

**Comment by JeffML** [comment]
I'm used Netlify's create-react-app script to create the project and dependencies, and I'm not going to mess around with the default configuration.

I'm going to close this.
